File: 0485-max-consecutive-ones/solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_count = 0
        current_count = 0
        
        for num in nums:
            if num:
                current_count += 1
                if current_count > max_count:
                    max_count = current_count
            else:
                current_count = 0
        
        return max(max_count, current_count)

File: 0485-max-consecutive-ones/solution.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max_len{0}, curr_len{0};
        for (const int& num: nums)
        {
            if (num) curr_len++;
            else
            {
                if (curr_len > max_len) max_len = curr_len;
                curr_len = 0;
            }
        }
        return max(curr_len, max_len);
    }
};
